diff --git a/utils/environment.sh b/utils/environment.sh
index 6e246cf..687e418 100644
--- a/utils/environment.sh
+++ b/utils/environment.sh
@@ -48,11 +48,3 @@ export XRAY_VIVADO="${XRAY_UTILS_DIR}/vivado.sh"
 
 # Verify an approved version is in use
 export XRAY_VIVADO_SETTINGS="${XRAY_VIVADO_SETTINGS:-/opt/Xilinx/Vivado/2017.2/settings64.sh}"
-# Vivado v2017.2 (64-bit)
-if [ "$(${XRAY_VIVADO} -h |grep Vivado |cut -d\  -f 2)" != "v2017.2" ] ; then
-    echo "Requires Vivado 2017.2. See https://github.com/SymbiFlow/prjxray/issues/14"
-    # Can't exit since sourced script
-    # Trash a key environment variable to preclude use
-    export XRAY_DIR="/bad/vivado/version"
-    return
-fi
